Migrate "Using metadata" from Sanity to MDX (#260)
clerkinc/clerk-docs: docs/organizations/metadata.mdx

---
title: Organization Metadata

> description: >-
>   Organization objects hold a set of metadata that can be used internally to store arbitrary information.

Reviewer: ```suggestion
description: Organization objects hold a set of metadata that can be used internally to store arbitrary information.
```

---

> import { Tables } from "@/components/Table";

Reviewer: Imports are unnecessary! The docs get rendered by our marketing repo, which is where the components live - they get rendered by the marketing repo.

```suggestion
import { Tables } from "@/components/Table";
```

# Organization Metadata

> **Organization objects hold a set of metadata that can be used internally to store arbitrary information.**
> 
> ## Overview
> 
> Metadata allows for custom data to be saved on the Organization object, Organization Invitation object, and Organization Membership object. There are two types of metadata: "**public**", and "**private**". Choosing which metadata field you should use for setting custom attributes on your organizations becomes a matter of access and visibility.

Reviewer: ```suggestion
Metadata allows for custom data to be saved on the `Organization` object, `Organization Invitation` object, and `Organization Membership` object. There are two types of metadata: **public** and **private**. Choosing which metadata field you should use for setting custom attributes on your organizations becomes a matter of access and visibility.
```

<Tables
  headings={["Metadata", "Frontend API", "Backend API"]}

>   headingsMeta={[{maxWidth:'200px'},{maxWidth: '200px'},{maxWidth: '200px'}]}

Reviewer: headingsMeta is a deprecated prop. we handle styling of components internally.

```suggestion
```

  rows={[
    {
    cells: [
      <>private</>, 
      <>-</>,
      <>Read/Write</>
    ],
    },
    {
      cells: [
        <>public</>,
        <>Read</>,
        <>Read/Write</>,
      ],
    },
  ]}
/>

<Callout type="info" emoji="ℹ️">
    Please note that Organization Invitations can only include **public** metadata.
</Callout>

<Callout type="warning">
    Metadata is limited to **8kb** maximum.
</Callout>

> ## Backend metadata

Reviewer: Let's remove this heading, I'm not convinced it's even necessary.

```suggestion
```

Both public and private metadata are set from the [Backend API](https://clerk.com/docs/reference/backend-api), but public metadata can be accessed from the [Frontend API](https://reference.clerk.dev/reference/frontend-api-reference) and [Backend API](https://clerk.com/docs/reference/backend-api). These types of metadata can be used to access non-sensitive information about an organization, such as their profile picture or their name. They can be used to build UIs where an organization might not be signed in, but you still want to show some of their profile info.

## Organization Metadata

### Public metadata

> The **publicMetadata** property should be used if you need to set some metadata from your back-end and have them displayed as read-only on the front-end.

Reviewer: This is going to be implemented moving forward, but we will use backend and frontend without dashes.

```suggestion
The `publicMetadata` property should be used if you need to set some metadata from your backend and have them displayed as read-only on the frontend.
```

> ```ts copy

Reviewer: copy is used if you want to include a filename prop

```suggestion
```ts
```

const updateOrg = await organizations.updateOrganizationMetadata(OrgID,{publicMetadata})
```

### Private metadata

> The **privateMetadata** property should be used if custom attributes contain sensitive information that should not be displayed on the front-end.

Reviewer: ```suggestion
The `privateMetadata` property should be used if custom attributes contain sensitive information that should not be displayed on the frontend.
```

> ```ts copy

Reviewer: ```suggestion
```ts
```

const updateOrg = await organizations.updateOrganizationMetadata(OrgID,{privateMetadata})
```

## Organization Membership Metadata

### Public metadata

> The **publicMetadata** property should be used if you need to set some metadata from your back-end and have them displayed as read-only on the front-end.

Reviewer: ```suggestion
The `publicMetadata` property should be used if you need to set some metadata from your back-end and have them displayed as read-only on the frontend.
```

### Private metadata

> The **privateMetadata** property should be used if custom attributes contain sensitive information that should not be displayed on the front-end.

Reviewer: ```suggestion
The `privateMetadata` property should be used if custom attributes contain sensitive information that should not be displayed on the frontend.
```

```ts copy
const updateOrg = await organizations.updateOrganizationMembershipMetadata(OrgID,{privateMetadata})
```

## Organization Invitation Metadata

### Public metadata

> The **publicMetadata** property should be used if you need to set some metadata from your back-end and have them displayed as read-only on the front-end.

Reviewer: ```suggestion
The `publicMetadata` property should be used if you need to set some metadata from your back-end and have them displayed as read-only on the frontend.
```
